### PR TITLE
MM-61148 Rewrite table parsing and improve error handling around Markdown code

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
The changes in https://github.com/mattermost/commonmark.js/pull/20 heavily rewrite the way that tables are parsed and are to fix the specific issue in this ticket, but I also went ahead and added some better error handling to the Markdown component to help prevent any figure issues from crashing the app.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61148

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates
- [x] Have run E2E tests by adding label `E2E iOS tests for PR`.

#### Device Information
This PR was tested on: None

#### Release Note
```release-note
Fixed crash caused by incorrect Markdown handling
```

### CODE REVIEW

File: app/components/markdown/markdown.tsx [hunk 1]
```diff
@@ -606,34 +611,76 @@ const Markdown = ({
...
-            node.appendChild(editIndicatorNode);
 
-            ast.appendChild(node);
+    let output;
+    try {
+        output = renderer.render(ast);
+    } catch (e) {
+        if (!errorLogged.current) {
+            // eslint-disable-next-line no-console
+            console.error(e);
```
Comment (enahum) [inline app/components/markdown/markdown.tsx:669]: Do use logError instead

File: package.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     "@stream-io/flat-list-mvcp": "0.10.3",
     "@voximplant/react-native-foreground-service": "3.0.2",
     "base-64": "1.0.0",
-    "commonmark": "npm:@mattermost/commonmark@0.30.1-3",
+    "commonmark": "github:mattermost/commonmark.js#bdd88ab24a012d0cff75708d7b61fcfc499366f7",
```
Comment (enahum) [inline package.json:46]: To update once released to npm

**Review by enahum — COMMENTED** [review]
Can you please link the changes to commonmark?

File: package.json (hunk 2 above)
Comment (hmhealey) [reply package.json:46]: Yep, that's the plan. We also have to merge that PR first

**Review by enahum — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
The app didnt crashing viewing posts with single pipe markdown posts


one thing (might be existing), markdows wont show up in threads view 

web view
image

root pos t with markdown and replies in mobile

WhatsApp Image 2024-11-08 at 1 07 04 PM

threads view of the root post with markdown in mobile

WhatsApp Image 2024-11-08 at 1 07 16 PM

**Pull request merged**